vic/drivers/image/src/vic_mpi_support.c

```diff
@@ -271,6 +273,134 @@ create_MPI_global_struct_type(MPI_Datatype *mpi_type)
 }
 
 /******************************************************************************
+ * @brief   Create an MPI_Datatype that represents the filenames_struct
+ * @details This allows MPI operations in which the entire filenames_struct
+ *          can be treated as an MPI_Datatype. NOTE: This function needs to be
+ *          kept in-sync with the global_param_struct data type in vic_def.h.
+ *
+ * @param mpi_type MPI_Datatype that can be used in MPI operations
+ *****************************************************************************/
+void
+create_MPI_filenames_struct_type(MPI_Datatype *mpi_type)
+{
+    int           nitems; // number of elements in struct
+    int           status;
+    int          *blocklengths;
+    size_t        i;
+    MPI_Aint     *offsets;
+    MPI_Datatype *mpi_types;
+
+    // nitems has to equal the number of elements in global_param_struct
+    nitems = 14;
+    blocklengths = (int *) malloc(nitems * sizeof(int));
```

Comment:
Change to 

```C
    blocklengths = malloc(nitems * sizeof(*blocklengths));
```

(check syntax)

Should update the remainder as well to be consistent with the issue #247